app/src/main/java/at/roteskreuz/stopcorona/screens/debug/exposure_notifications/DebugExposureNotificationsViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package at.roteskreuz.stopcorona.screens.debug.exposure_notifications
+
+import android.app.Activity
+import android.app.Application
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.lifecycle.AndroidViewModel
+import at.roteskreuz.stopcorona.R
+import at.roteskreuz.stopcorona.skeleton.core.model.helpers.DataState
+import at.roteskreuz.stopcorona.skeleton.core.model.helpers.DataStateObserver
+import at.roteskreuz.stopcorona.skeleton.core.utils.observeOnMainThread
+import at.roteskreuz.stopcorona.utils.NonNullableBehaviorSubject
+import com.google.android.apps.exposurenotification.nearby.ExposureNotificationClientWrapper
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.Status
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationStatusCodes
+import io.reactivex.Observable
+import timber.log.Timber
+
+class DebugExposureNotificationsViewModel(
+    application: Application
+    ): AndroidViewModel(application){
+    private val exposureNotificationsEnabledSubject = NonNullableBehaviorSubject<Boolean>(false);
+    private val exposureNotificationsErrorSubject = NonNullableBehaviorSubject<String>("no error");
+    private val exposureNotificationsErrorState = DataStateObserver<Status>()
+
+
+    //TODO: move to a ExposureNotificationsRepository
+    //TODO: get inspired by https://github.com/austrianredcross/stopp-corona-android/blob/develop/app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DiscoveryRepository.kt
```

Comment:
Let's merge it to story, but we need to prepare the repository before merging story to develop.